HW1/P4/P4.py

```diff
@@ -0,0 +1,32 @@
+from P4_bfs import *
+import findspark
+findspark.init()
+import pyspark
+
+def preprocess_data( filename, sc ):
+    textFile = sc.textFile( filename )
+    textFile.cache()
+    textFile_pre_process = textFile.map( lambda x: ( x.split( "\"" )[3], x.split( "\"" )[1] ) ).partitionBy(num_of_partitions)
```

Comment:
In Python, single quotes (') are your friend when you actually want to match on double quotes and vice versa